### PR TITLE
feat: add process definition search to client

### PR DESCRIPTION
**Pull request by ralfpuchert**
## Description

* Add process definition search to client
* Based on https://github.com/camunda/camunda/pull/22914 (service, controller)

## Related issues

closes #

### CODE REVIEW

File: clients/java/src/main/java/io/camunda/zeebe/client/api/search/filter/ProcessDefinitionFilter.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+public interface ProcessDefinitionFilter extends SearchRequestFilter {
+
+  /**
+   * Filters process definitions by the specified process definition key.
+   *
+   * @param processDefinitionKey the key of the process definition
+   * @return the updated filter
+   */
+  ProcessDefinitionFilter processDefinitionKey(final Long processDefinitionKey);
```
Comment (houssain-barouni) [inline clients/java/src/main/java/io/camunda/zeebe/client/api/search/filter/ProcessDefinitionFilter.java:28]: Minor: can we use a primitive long ? that avoids to handle the null case

File: clients/java/src/main/java/io/camunda/zeebe/client/api/search/filter/ProcessDefinitionFilter.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+   */
+  ProcessDefinitionFilter resourceName(final String resourceName);
+
+  /**
+   * Filters process definitions by the specified version.
+   *
+   * @param version the version of the process definition
+   * @return the updated filter
+   */
+  ProcessDefinitionFilter version(final Integer version);
```
Comment (houssain-barouni) [inline clients/java/src/main/java/io/camunda/zeebe/client/api/search/filter/ProcessDefinitionFilter.java:52]: same here, I think it is better to use a primitive int. It indicates to the user that null is not a valid value here.

File: clients/java/src/main/java/io/camunda/zeebe/client/api/search/response/ProcessDefinition.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.camunda.zeebe.client.api.search.response;
+
+public interface ProcessDefinition {
```
Comment (houssain-barouni) [inline clients/java/src/main/java/io/camunda/zeebe/client/api/search/response/ProcessDefinition.java:18]: This looks very similar to `io.camunda.zeebe.client.api.response.Process` that is used in the command clients. Should we use 2 different datamodels?

**Review by houssain-barouni — APPROVED** [review]
LGTM overall,
I had some questions , these questions apply to other client implementations as well.
Also there is a TODO [here](https://github.com/camunda/camunda/blob/8c04534ab1d0912150b332f2f28311dfa6058dda/qa/integration-tests/src/test/java/io/camunda/it/client/ProcessQueryTest.java#L497) that was expecting the changes in your pull request, so maybe we can fix it here.

**Review by nathansandi — APPROVED** [review]
Besides the minor suggestions from @houssain-barouni  which I am agree 
LGTM, good job 👍

File: clients/java/src/main/java/io/camunda/zeebe/client/api/search/response/ProcessDefinition.java (hunk 3 above)
Comment (ralfpuchert) [reply clients/java/src/main/java/io/camunda/zeebe/client/api/search/response/ProcessDefinition.java:18]: Good catch! Will be handled in https://github.com/camunda/camunda/issues/23120

**Pull request merged**